### PR TITLE
Add Park.generateGuest() to plugin API

### PR DESCRIPTION
**Pull request by kendfrey**
Implements #22848.

### CODE REVIEW

**Comment by kendfrey** [comment]
This is my first PR to OpenRCT2 so don't be afraid to point out anything I missed.

One question: I wondered about making the function return the guest ID, rather than the guest object itself. This would be less convenient but maybe there is a compelling reason to do it instead.

File: distribution/openrct2.d.ts [hunk 1]
```diff
@@ -3807,6 +3807,12 @@ declare global {
          */
         readonly guestGenerationProbability: number;
 
+        /**
+         * Spawns a new guest at the edge of the map.
```
Comment (tupaschoal) [inline distribution/openrct2.d.ts:3811]:
```suggestion
         * Spawns a new guest at a random peep spawn point.
```
This is technically more accurate.

**Comment by ZehMatt** [comment]
> This is my first PR to OpenRCT2 so don't be afraid to point out anything I missed.
> 
> One question: I wondered about making the function return the guest ID, rather than the guest object itself. This would be less convenient but maybe there is a compelling reason to do it instead.

I think returning the object is fine. I think we raise the plugin API version after changes, also feel free to add yourself to the list of contributors if you like that. This will also need a changelog entry.

**Review by tupaschoal — COMMENTED** [review]
This looks fine. Need to add a changelog entry as well, aside from what @ZehMatt mentioned.

PS: I'm wondering whether we can get away with not having `ScPark` depend on `ScGuest` or vice-versa

**Review by ZehMatt — APPROVED** [review]

**Pull request merged**